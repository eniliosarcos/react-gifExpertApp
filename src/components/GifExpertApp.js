import React, { useState } from "react";
import "./GifExpertApp.css";
import AddCategory from "./AddCategory/AddCategory";
import GifCollection from "./GifCollection/GifCollection";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Dragon ball"]);

  return (
    <div>
      <h2>Gif Expert App</h2>
      <AddCategory setCategory={setCategories}></AddCategory>
      <hr></hr>

      <div className="collection-content">
        {categories.map((category) => {
          return (
            <GifCollection key={category} collection={category}></GifCollection>
          );
        })}
      </div>
    </div>
  );
};

export default GifExpertApp;
