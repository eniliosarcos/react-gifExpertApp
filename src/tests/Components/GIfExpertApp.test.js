import { shallow } from "enzyme";
import GifExpertApp from "../../components/GifExpertApp";


describe('Prueba con el componente GifExpertApp', () => {

    test('Debe de mostrar el componente correctamente', () => {
        const wrapper = shallow(<GifExpertApp></GifExpertApp>);
        expect(wrapper).toMatchSnapshot();
    })
    

    test('debe de mostrar una lista de categorias', () =>{
        const categories = ['One punch', 'Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}></GifExpertApp>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifCollection').length).toBe(categories.length);
    })
})
