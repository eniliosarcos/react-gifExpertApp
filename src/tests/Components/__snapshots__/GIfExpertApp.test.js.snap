// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con el componente GifExpertApp Debe de mostrar el componente correctamente 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <div
    className="collection-content"
  />
</div>
`;

exports[`Prueba con el componente GifExpertApp debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <div
    className="collection-content"
  >
    <GifCollection
      collection="One punch"
      key="One punch"
    />
    <GifCollection
      collection="Dragon ball"
      key="Dragon ball"
    />
  </div>
</div>
`;
